ListApp/Sources/ListApp.Contracts/Interfaces/IRepository.cs

```diff
@@ -0,0 +1,52 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using ListApp.Contracts.Models;
+
+namespace ListApp.Contracts.Interfaces
+{
+    public interface IRepository
+    {
+        /// <summary>
+        /// A method that returns all keys from repository
+        /// </summary>
+        /// <returns>A collection of all keys</returns>
+        Task<IEnumerable<Guid>> GetKeysAsync();
+
+        /// <summary>
+        /// A method that returns all items in the repository.
+        /// </summary>
+        /// <returns>A collection of all items</returns>
+        Task<IEnumerable<ListItem>> GetAllAsync();
+
+        /// <summary>
+        /// A method that returns a specific item identified by given key.
+        /// </summary>
+        /// <param name="key"></param>
+        /// <returns>The item with given key, null if no such item exists.</returns>
+        Task<ListItem> GetAsync(Guid key);
+
+        /// <summary>
+        /// Stores given entity in the repository.
+        /// </summary>
+        /// <param name="entity">The entity to store in the repository</param>
+        /// <exception cref="ArgumentNullException">if a null argument is passed</exception>
+        /// <exception cref="DuplicateKeyException">if an item with given key already exists</exception>
+        Task<ListItem> AddAsync(ListItem entity);
```

Comment:
The item you're adding is just an "item". The "entity" seems needlessly abstract.